### ISSUE TITLE
Global option "--help" works as intended?

### ISSUE THREAD

**Issue by CrystalMethod** [suzuki-shunsuke/cmdx#388]
As a user who is not yet familiar with a new tool, I usually start by running the tool with the "-h" or "--help" option to learn more about how to use the tool.

```
$ cmdx --help
Incorrect Usage. flag: help requested

cmdx - task runner
https://github.com/suzuki-shunsuke/cmdx

Configuration file isn't found.
First of all, let's create a configuration file.

$ cmdx --init

Or if the configuration file already exists but the file path is unusual, please specify the path by --config (-c) option.

$ cmdx -c <YOUR_CONFIGURATION_FILE_PATH> <COMMAND> ...
flag: help requested
```

But the configuration actually exists.

```console
$ ls .cmdx.yaml
.cmdx.yaml
```

Let's try `cmdx` without any option:
```
$ cmdx
NAME:
   cmdx - task runner

USAGE:
   cmdx [global options] command [command options] [arguments...]

VERSION:
   1.6.1 (633ecaa2ee57aa308c05bf62546e6da5639ac764)

AUTHOR:
   Shunsuke Suzuki

COMMANDS:
   update, u  Update project tooling
   help, h    Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --config value, -c value       configuration file path [$CMDX_CONFIG_PATH]
   --name value, -n value         configuration file name. The configuration file is searched from the current directory to the root directory recursively
   --working-dir value, -w value  The working directory path. By default, the task is run on the directory where the configuration file is found [$CMDX_WORKING_DIR]
   --init, -i                     create the configuration file (default: false)
   --list, -l                     list tasks (default: false)
   --quiet, -q                    don't output the executed command (default: false)
   --dry-run, -d                  output the script but don't run it actually (default: false)
   --help, -h                     show help (default: false)
   --version, -v                  print the version (default: false)
```

Yes, there ist a global `--help` option. So lets test global option `--version`:

```
$ cmdx --version
cmdx version 1.6.1 (633ecaa2ee57aa308c05bf62546e6da5639ac764)
```

For me it looks rather unusual that the global option `--help` ends with an error complaining that the configuration file would be missing ... especially since this file is actually present.

**Comment by suzuki-shunsuke** [comment]
Thank you for your report.
The help message was fixed in https://github.com/suzuki-shunsuke/cmdx/pull/322 but this fix isn't released yet.
I don't remember the background of this issue because recently I don't maintain this tool.
I'll take a look and consider releasing a new version.

**Comment by suzuki-shunsuke** [comment]
`cmdx help` (not `--help`) works as expected.

**Comment by suzuki-shunsuke** [comment]
Let me note the description about this issue.
The behaviour of `cmdx` is different from usual CLI because `cmdx` is a task runner and sub commands are defined in configuration file.
Sub command including `help` is evaluated after reading configuration file, though `--help` option is evaluated before reading configuration file because `--help` is a built in option of `urfave/cli` at the moment.
That's why the behaviour of `--help` and `help` is different.

`--help` is help message of `cmdx` itself, on the other hand `help` command is help message of user defined tasks.
But this difference is not intuitive and confusing users.
Most users would expect `--help` and `help` are same.

I'll consider the issue.

**Comment by suzuki-shunsuke** [comment]
Released v1.7.0.
https://github.com/suzuki-shunsuke/cmdx/releases/tag/v1.7.0

The message was improved. https://github.com/suzuki-shunsuke/cmdx/pull/322#issue-1087371694

```console
$ cmdx --help
cmdx - task runner
https://github.com/suzuki-shunsuke/cmdx

Please run "cmdx help" to show help.
```

From this version, pre built binaries for arm64 have also been released.
aqua Standard Registry will be fixed according to this change. https://github.com/aquaproj/aqua-registry/pull/10762

As I mentioned, we will consider improving the help message same as `help` command.

**Comment by CrystalMethod** [comment]
Many thanks. For me, this "quick fix" is sufficient at the moment.
